src/libs/Middleware/Reauthentication.js

```diff
@@ -58,18 +58,14 @@ function Reauthentication(response, request, isFromSequentialQueue) {
                 // There are some API requests that should not be retried when there is an auth failure like
                 // creating and deleting logins. In those cases, they should handle the original response instead
                 // of the new response created by handleExpiredAuthToken.
-                const shouldRetry = lodashGet(request, 'data.shouldRetry');
+                // If the request was from the sequential queue we don't want to return, we want to run the reauthentication callback and retry the request
                 const apiRequestType = lodashGet(request, 'data.apiRequestType');
 
                 // For the SignInWithShortLivedAuthToken command, if the short token expires, the server returns a 407 error,
                 // and credentials are still empty at this time, which causes reauthenticate to throw an error (requireParameters),
                 // and the subsequent SaveResponseInOnyx also cannot be executed, so we need this parameter to skip the reauthentication logic.
                 const skipReauthentication = lodashGet(request, 'data.skipReauthentication');
-                if ((!shouldRetry && !apiRequestType) || skipReauthentication) {
-                    if (isFromSequentialQueue) {
-                        return data;
-                    }
```

Comment:
> In particular, LogOut is a weird one. It should be in the SequentialQueue since it's an API.write() but looking at the logic we are changing here and I'm not seeing the code path where will avoid the call to reauthenticate()

I think we should call reauthenticate for LogOut. We need a valid authToken to call the command. So we get that first then call it if the current one is expired

Reply:
> e.g. in those places we should be able to handle the jsonCode by waiting for the promise to resolve. But it mainly only look relevant to this call...

Hmm, yeah I think this one is wrong, if we have an invalid authToken why wouldn't we just get a new one and call the command again?

Reply:
>I think we should call reauthenticate for LogOut

That makes sense to me. Ideally, we should [delete the temporary login that we created](https://github.com/Expensify/Web-Expensify/blob/a3113f14e94524fa6d66b87d22f9ec5fd0bb1dd1/api.php#L2482-L2484).

Reply:
Ok, so for the deep link one... I _think_ it's fine either way because the param actually doesn't do anything 👍 

>Hmm, yeah I think this one is wrong, if we have an invalid authToken why wouldn't we just get a new one and call the command again?

`AuthenticatePusher` - we might want to create a follow up issue to make sure we are handling that case correctly or cleaning up any dead flows.

I log searched for:

```
[PusherAuthorizer] Unable to authenticate Pusher because authToken is expired
```

and it doesn't happen anymore (as expected) so we can probably remove these lines...

https://github.com/Expensify/App/blob/2beee7be3981debf992d1a43d0a8d6409126b5c2/src/libs/actions/Session/index.js#L553-L560

NAB but we should do this clean up.